channel.go

```diff
@@ -299,6 +313,9 @@ func (ch *Channel) Serve(l net.Listener) error {
 	mutable.peerInfo.IsEphemeral = false
 	ch.log = ch.log.WithFields(LogField{"hostPort", mutable.peerInfo.HostPort})
 
+	// Start the idle connection timer.
+	ch.mutable.idleSweep.Start()
```

Comment:
Channels are re-usable after Close(). So we need to have something to start the idle_sweep again if the user decides to re-use the channel. The alternative is to prevent channels from being reused, but that might break some existing code.

Reply:
Hmm I don't think we allow channels to be reused. `Serve` ensures the channel state is `ChannelClient` (which is the initial state), and we don't allow going backwards anywhere.